lib/collection/src/collection_manager/collection_updater.rs

```diff
@@ -42,31 +42,33 @@ impl CollectionUpdater {
         operation: CollectionUpdateOperations,
         hw_counter: &HardwareCounterCell,
     ) -> CollectionResult<usize> {
-        // Allow only one update at a time, ensure no data races between segments.
-        // let _lock = self.update_lock.lock().unwrap();
-        let scroll_lock = segments.read().scroll_read_lock.clone();
-
         // Use block_in_place here to avoid blocking the current async executor
-        let _scroll_lock = tokio::task::block_in_place(|| scroll_lock.blocking_write());
+        tokio::task::block_in_place(|| {
+            // Allow only one update at a time, ensure no data races between segments.
+            // let _lock = self.update_lock.lock().unwrap();
 
-        let operation_result = match operation {
-            CollectionUpdateOperations::PointOperation(point_operation) => {
-                process_point_operation(segments, op_num, point_operation, hw_counter)
-            }
-            CollectionUpdateOperations::VectorOperation(vector_operation) => {
-                process_vector_operation(segments, op_num, vector_operation, hw_counter)
-            }
-            CollectionUpdateOperations::PayloadOperation(payload_operation) => {
-                process_payload_operation(segments, op_num, payload_operation, hw_counter)
-            }
-            CollectionUpdateOperations::FieldIndexOperation(index_operation) => {
-                process_field_index_operation(segments, op_num, &index_operation, hw_counter)
-            }
-        };
+            let scroll_lock = segments.read().scroll_read_lock.clone();
+            let _scroll_lock = scroll_lock.blocking_write();
+
+            let operation_result = match operation {
+                CollectionUpdateOperations::PointOperation(point_operation) => {
+                    process_point_operation(segments, op_num, point_operation, hw_counter)
+                }
+                CollectionUpdateOperations::VectorOperation(vector_operation) => {
+                    process_vector_operation(segments, op_num, vector_operation, hw_counter)
+                }
+                CollectionUpdateOperations::PayloadOperation(payload_operation) => {
+                    process_payload_operation(segments, op_num, payload_operation, hw_counter)
+                }
+                CollectionUpdateOperations::FieldIndexOperation(index_operation) => {
+                    process_field_index_operation(segments, op_num, &index_operation, hw_counter)
+                }
+            };
 
-        CollectionUpdater::handle_update_result(segments, op_num, &operation_result);
+            CollectionUpdater::handle_update_result(segments, op_num, &operation_result);
 
-        operation_result
+            operation_result
+        })
```

Comment:
`CollectionUpdate::update` is blocking, but we call it from async context in a few places. Seems reasonable to wrap the whole `update` into `block_in_place`.

`block_in_place` does have a performance hit when called from async, but
1) we already use `block_in_place` inside `update`, so we already "pay" for this perf hit, we just extend `block_in_place` scope here, which may actually improve perf when called from async
2) `update_worker_fn` (which is the most important `update` path) is a blocking task now, and other places where we still call `update` from async are less critical (e.g., `LocalShard::load_from_wal`)